source/src/BNFC/MultiView.hs

```diff
@@ -39,17 +39,14 @@ preprocessMCF f = do
 
 extract :: String -> LGrammar -> [(FilePath, Grammar)]
 extract name (LGr ldefs) =
-  [(file lang,Grammar [unldef ldef | ldef <- ldefs, isFor lang ldef]) |
+  [(file lang,Grammar $ concatMap (getDef lang) ldefs) |
```

Comment:
Alternatively, `mapMaybe`http://hackage.haskell.org/package/base-4.12.0.0/docs/Data-Maybe.html#v:mapMaybe can be used for that purpose; then `getDef` would return either `Just d` or `Nothing`.